src/haddock/gear/prepare_run.py

```diff
@@ -517,3 +530,113 @@ def populate_mol_parameters(modules_params):
                 defaults[param],
                 )
     return
+
+
+def check_if_path_exists(path):
+    """
+    Check if a path exists and raises an error if it does not exist.
+
+    For example given this path "../config/project_01/file.txt" it would find
+    the following path "../config/project-01".
+
+    Parameters
+    ----------
+    path : AnyStr | PathLike
+        The path to check.
+
+    Returns
+    -------
+    None
+        If the path does exist.
+
+    Raises
+    ------
+    ValueError
+        If the path does not exist.
+    """
+    path = os.path.normpath(path)
+    if os.path.exists(path):
+        return None
+
+    split = os.path.split(path)
+    elements = split[0].split(os.sep) + split[1]
+    reconstituted_path = ""
+    error = ("", "", "")
+    for part in elements:
+        next_folder = reconstituted_path + os.sep + part
+        if not os.path.exists(next_folder):
+            error = (reconstituted_path, fuzzy_match([part],
+                     os.listdir(reconstituted_path))[0][1], part)
+            break
+
+    msg = (f"The following file could not be found: \'{path}\'."
+           f"In the folder \'{error[0]}\' the following \'{error[1]}\' "
+           f"is the closest match to the supplied \'{error[2]}\', did "
+           "you mean to open this?")
+    raise ValueError(msg)
+
+
+def fuzzy_match(user_input, possibilities):
+    """
+    Find the closest possibility to the user supplied input.
+
+    Parameters
+    ----------
+    user_input : list(string)
+        List of strings with the faulty input given by the user.
+    possibilities : list(string)
+        List of strings with all possible options that would be
+        valid in this context.
+
+    Returns
+    -------
+    list(string, string)
+        The closest string from the possibilities to each string of the
+        `user_input`. With as first element of the tuple the user_input
+        string, and as second element the matched possibility.
+    """
+    results = list()
+
+    for user_word in user_input:
```

Comment:
You use our nice function:

```
from haddock.libs.libutil import transform_to_list
....
for user in transform_to_list(user_input):
...
```
like that you can call the function directly with a string.